src/main/java/org/jvnet/hudson/update_center/Plugin.java

```diff
@@ -379,11 +380,21 @@ public String getScmUrl() throws IOException {
         return labels.split("\\s+");
     }
 
-    /** @return The plugin name defined in the POM &lt;name>; falls back to the wiki page title, then artifact ID. */
+    /** @return The plugin name defined in the POM &lt;name> modified by simplication rules (no 'Jenkins', no 'Plugin'); then artifact ID. */
     public String getName() throws IOException {
         String title = selectSingleValue(getPom(), "/project/name");
         if (title == null) {
             title = artifactId;
+        } else {
+            title = StringUtils.removeStart(title.trim(), "Jenkins ");
+            title = StringUtils.removeStart(title, "Hudson ");
+            title = StringUtils.removeEnd(title, " for Jenkins");
+            title = StringUtils.removeEnd(title, " Plugin");
+            title = StringUtils.removeEnd(title, " plugin");
+            title = StringUtils.removeEnd(title, " Plug-In");
+            title = StringUtils.removeEnd(title, " Plug-in");
+            title = StringUtils.removeEnd(title, " plug-in");
+            title = StringUtils.removeEnd(title, " Jenkins");
```

Comment:
Dunno exactly where each value come from, but in the list from @orrc there is also in the middle column (seems to be the one that is used here) : 
- ClearCase UCM **Plugin!** 
- Computer-queue**-plugin**
  - leiningen**-plugin**
- Cucumber json test reporting**.**  (dot could make the listing a bit weird)
  - Inedo BuildMaster plugin for Jenkins**.**
- Mail Commander **Plugin for Jenkins-ci**
- Maven Metadata **Plugin for Jenkins CI server**
- PVCS SCM **plugin for Hudson**
- Seapine Surround SCM **- Jenkins Plugin** (after removal of Jenkins Plugin, it remains just `Seapine Surround SCM -` )
- HudsonWindmill**Plugin**

Reply:
Current data is `jq -r '.plugins[] | .title'` of http://updates.jenkins-ci.org/update-center.actual.json

PVCS SCM and Seapine Surround SCM cleaned up their names enough so it looks reasonable. I couldn't find HudsonWindmillPlugin.

I'm fine with the rest looking weird and different, I doubt we'll see any of these a lot. In some cases, even their authors clearly didn't care what the display name would look like, so why should we?

https://plugins.jenkins.io/maven-metadata-plugin (~2700 installs)
https://plugins.jenkins.io/mailcommander (487)
https://plugins.jenkins.io/computer-queue-plugin (142)
https://plugins.jenkins.io/leiningen-plugin (191)
https://plugins.jenkins.io/inedo-buildmaster (15)
https://plugins.jenkins.io/inedo-proget (40)

Given that we won't be able to magically remove all redundant bits and stupid "formatting" like trailing periods that people come up with, I don't think rules as complex as those proposed by @orrc are useful.

The following would look weird under the current proposed rules:

> PTC Integrity CM - Jenkins Plugin
> Plugin Usage - Plugin

I plan to amend this to address the problem of trailing punctuation, but that's it.

Reply:
Works perfectly for me 👍

Reply:
> Given that we won't be able to magically remove all redundant bits and stupid "formatting" like trailing periods that people come up with, I don't think rules as complex as those proposed by @orrc are useful.

My code just took that into account with `.replaceAll("[ -.!]+$", "")`, and I don't see any other weird suffixes in the current titles.

> In the mean time I'll take your reference to "I proposed this years ago" to mean approval ;-)

Yeah, I definitely agree with the idea.  There was a dev list thread about this, but not sure why I ended up not opening a PR.